src/app/icd/ICDMonitoringTable.h

```diff
@@ -59,6 +59,26 @@ struct ICDMonitoringEntry : public PersistentData<kICDMonitoringBufferSize>
     CHIP_ERROR Serialize(TLV::TLVWriter & writer) const override;
     CHIP_ERROR Deserialize(TLV::TLVReader & reader) override;
     void Clear() override;
+    /**
+     * @brief Set the Key object
+     *        This method will create a new keyHandle. The key handle might contain either
+     *        the raw key or a keyID depending on which Crypto implementation is used.
+     *        To avoid leaking keys, API consumers must either call the DeleteKey method
+     *        or save the entry within the ICDMonitoring Table before this object goes out of scope.
+     *
+     *       Calling SetKey() on an ICDMonitoringEntry that already has a valid key handle will lead to
+     *       the old key being deleted if the key handle is a key id, not raw data.  This can lead to keys
+     *       that were already persisted via the ICDMonitoring table getting lost.
```

Comment:
I don't think this comment it true anymore.

Reply:
removed